### PR TITLE
Make symfony bundle optional

### PR DESCRIPTION
**Pull request by jkrzefski**
I moved all the symfony bundle specific files into a new namespace `Bundle`. Also I removed the requirement for `symfony/framework` and instead added the necessary symfony components as dev-dependencies and suggestions.

See this conversation: https://twitter.com/jkrzefski/status/1402881774535970817

### CODE REVIEW

**Comment by adrienbrault** [comment]
Hey, this is great :)

What about having the following namespace structure instead:
- `Sourceability\Instrumentation\`
  - `Sourceability\Instrumentation\Profiler\`
  - `Sourceability\Instrumentation\EventListener\` 
  - `Sourceability\Instrumentation\Bundle\`
    - `Sourceability\Instrumentation\Bundle\DependencyInjection`

Also renaming:
- the composer package to `sourceability/instrumentation`
- the repository to `sourceability/instrumentation`

File: src/Bundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+{
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder('sourceability_instrumentation');
+        $rootNode = $treeBuilder->getRootNode();
+
+        $rootNode
+            ->children()
+            ->arrayNode('profilers')
+            ->children()
```
Comment (adrienbrault) [inline src/Bundle/DependencyInjection/Configuration.php:20]: It looks like the indentation went away

Comment (jkrzefski) [reply src/Bundle/DependencyInjection/Configuration.php:20]: I ran `make cs-fix` and this was the result.

Comment (adrienbrault) [reply src/Bundle/DependencyInjection/Configuration.php:20]: Ah, understood, I believe the following has to be updated for `cs-fix` to not mess with the indentation: https://github.com/sourceability/instrumentation-bundle/blob/a877f6cb527c88d9d2121c87a202c104f35f3531/ecs.php#L31-L36

Comment (jkrzefski) [reply src/Bundle/DependencyInjection/Configuration.php]: Done.

**Comment by jkrzefski** [comment]
> Hey, this is great :)
> 
> What about having the following namespace structure instead:
> 
> * `Sourceability\Instrumentation\`
>   
>   * `Sourceability\Instrumentation\Profiler\`
>   * `Sourceability\Instrumentation\EventListener\`
>   * `Sourceability\Instrumentation\Bundle\`
>     
>     * `Sourceability\Instrumentation\Bundle\DependencyInjection`
> 
> Also renaming:
> 
> * the composer package to `sourceability/instrumentation`
> * the repository to `sourceability/instrumentation`

I like this a lot. I just updated the PR to include these changes. But of course I cannot change the repository name nor can I change the name on packagist. I leave that to you. Don't forget the entries `homepage`, `support.issues` and `support.source` in the `composer.json` file.

**Comment by adrienbrault** [comment]
Thank you! I'm going to take care of the composer/packagist/repo last steps

**Comment by adrienbrault** [comment]
- https://packagist.org/packages/sourceability/instrumentation-bundle has been marked as `abandoned`
- https://packagist.org/packages/sourceability/instrumentation has been created
- `0.2.0` has been tagged
- the repository url has been updated to https://github.com/sourceability/instrumentation

Thank you again!

**Comment by jkrzefski** [comment]
Wow, that was fast 🚀
Thank you so much for your awesome work.

**Pull request merged**